src/v2/coins/abstractUtxoCoin.js

```diff
@@ -1121,13 +1132,30 @@ class AbstractUtxoCoin extends BaseCoin {
       const baseKeyPath = deriveKeys(deriveKeys(keys, 0), 0);
       const userKeyArray = deriveKeys(baseKeyPath, 0);
       const changeKeyArray = deriveKeys(baseKeyPath, 1);
+      const userKeyArraySW = deriveKeys(baseKeyPath, 10);
+      const changeKeyArraySW = deriveKeys(baseKeyPath, 11);
 
       // Collect the unspents
       const addressesById = {};
-      const userUnspents = yield queryBlockchainUnspentsPath(userKeyArray, '/0/0/0');
-      const changeUnspents = yield queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1');
-      const unspents = userUnspents.concat(changeUnspents);
-
+      // Gather the queries, to be executed in parallel
+      const includeP2SH_P2WSH = _.isUndefined(params.ignoreAddressTypes) || _.isEmpty(params.ignoreAddressTypes) || !_.includes(params.ignoreAddressTypes, AbstractUtxoCoin.AddressTypes.P2SH_P2WSH);
+      let queries = [
+        queryBlockchainUnspentsPath(userKeyArray, '/0/0/0', false),
+        queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1', false)]
+      if (includeP2SH_P2WSH) {
+        queries = queries.concat([queryBlockchainUnspentsPath(userKeyArraySW, '/0/0/10', true), queryBlockchainUnspentsPath(changeKeyArraySW, '/0/0/11', true)]);
+      }
+      const queryResponse = yield Promise.all(queries);
+      const userUnspents = queryResponse[0];
+      const changeUnspents = queryResponse[1];
+      let unspents, userUnspentsSW, changeUnspentsSW;
+      if (includeP2SH_P2WSH) {
+        userUnspentsSW = queryResponse[2];
+        changeUnspentsSW = queryResponse[3];
+        unspents = userUnspents.concat(changeUnspents).concat(userUnspentsSW).concat(changeUnspentsSW);
```

Comment:
.concat().concat().concat() seems strange.

Why not `userUnspents.concat([changeUnspents, userUnspentsSW, changeUnspentsSW])`